guava/src/com/google/common/escape/UnicodeEscaper.java

```diff
@@ -127,8 +134,10 @@ public String escape(String string) {
    * @throws IllegalArgumentException if the scanned sub-sequence of {@code csq} contains invalid
    *     surrogate pairs
    */
-  protected int nextEscapeIndex(CharSequence csq, int start, int end) {
-    int index = start;
+  @SuppressWarnings("compound.assignment.type.incompatible")//index += Character.isSupplementaryCodePoint(cp) ? 2 : 1 occurs
+  //only if escape(cp) == null( cp is not supplementaryCodePoint)
```

Comment:
It's not clear from this comment why the fact you stated implies that the code is safe from array bounds errors. Can you clarify this? For instance, it would be helpful to summarize here the effect of `Character#isSupplementaryCodePoint`, since you claim it is relevant to the proof.